neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -1596,20 +1606,24 @@ object Neo4jStoreSingleton : GraphStore {
                         it
                     }.linkReversalActionToTransaction(transaction) {
                         paymentProcessor.removeInvoice(it.id)
-                        logger.error(NOTIFY_OPS_MARKER,
-                                """Failed to create or pay invoice for customer ${customer.id}, invoice-id: ${it.id}.
+                        logger.warn(NOTIFY_OPS_MARKER,
+                                """Failed to pay invoice for customer ${customer.id}, invoice-id: ${it.id}.
                                    Verify that the invoice has been deleted or voided in Stripe dashboard.
                                 """.trimIndent())
                     }.bind()
 
             /* Force immediate payment of the invoice. */
             val invoicePaymentInfo = paymentProcessor.payInvoice(invoice.id)
                     .mapLeft {
-                        logger.error("Payment of invoice ${invoice.id} failed for customer ${customer.id}.")
+                        logger.warn("Payment of invoice ${invoice.id} failed for customer ${customer.id}.")
+                        /* Adds failed purchase to customer history. */
+                        AuditLog.warn(customerId = customer.id,
+                                message = "Failed to complete purchase of product $sku for $price.amount} ${price.currency} " +
+                                        "status: ${it.code} decline reason: ${it.declineCode}")
                         it
                     }.linkReversalActionToTransaction(transaction) {
                         paymentProcessor.refundCharge(it.chargeId)
-                        logger.error(NOTIFY_OPS_MARKER,
+                        logger.warn(NOTIFY_OPS_MARKER,
                                 """Refunded customer ${customer.id} for invoice: ${it.id}.
                                    Verify that the invoice has been refunded in Stripe dashboard.
                                 """.trimIndent())
```

Comment:
Move the starting `"""` to the line above.  Keeping it in the same line prevents timeIndent() from working.

Reference: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/trim-indent.html

Reply:
Fixed in update.